src/Scope.ts

```diff
@@ -359,7 +381,8 @@ export class Scope {
                             namespaces: new Map<string, NamespaceContainer>(),
                             classStatements: {},
                             functionStatements: {},
-                            statements: []
+                            statements: [],
+                            enumStatements: {}
```

Comment:
Let's use a `Map<string, EnumStatement>` here like we do for namespaces. (I intend on converting the class and function statements to using maps in the future)